ccpp/data/GFS_typedefs.F90

```diff
@@ -3567,7 +3637,11 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
     Model%fhzero           = fhzero
     Model%ldiag3d          = ldiag3d
     Model%qdiag3d          = qdiag3d
-    if (Model%qdiag3d .and. .not. Model%ldiag3d) then
+    if (ldiag3d .and. .not. lssav) then
```

Comment:
Strictly speaking this test is not needed. `lssav` may be `.false.` at init time, just because it is set to `.true.` in `GFS_time_vary_pre_timestep_init`. I know this is awkward, needs to be cleaned up in the near future. As long as line 3640 works for now, let's keep it.